### PR TITLE
Update Vm.sol with exists, isFile and isDir

### PR DESCRIPTION
**Pull request by ruvaag**

### CODE REVIEW

**Comment by ruvaag** [comment]
@Evalir PR for upstream changes, tagging for vis. There is no associated issue in this repo, but let me know if we need one for completeness.

File: src/Vm.sol [hunk 1]
```diff
@@ -541,4 +541,10 @@ interface Vm is VmSafe {
     function transact(uint256 forkId, bytes32 txHash) external;
     // Suspends execution of the main thread for `duration` milliseconds
     function sleep(uint256 duration) external;
+    // Returns true if the given path points to an existing entity, else returns false
+    function exists(string calldata path) external returns (bool);
+    // Returns true if the path exists on disk and is pointing at a regular file, else returns false
+    function isFile(string calldata path) external returns (bool);
+    // Returns true if the path exists on disk and is pointing at a directory, else returns false
+    function isDir(string calldata path) external returns (bool);
```
Comment (Evalir) [inline src/Vm.sol:549]: let's also name the return vars

**Review by Evalir — APPROVED** [review]
lgtm, albeit up to @mds1 for final review / approve! just a smol nit:

File: src/Vm.sol (hunk 1 above)
Comment (ruvaag) [reply src/Vm.sol]: Ack - fixed with reflexive named params for max clarity

**Review by mds1 — APPROVED** [review]
Thanks! I moved the new cheats from the `Vm` interface to the `VmSafe` interface next to the other filesystem cheats

**Pull request merged**